execution/src/worker.rs

```diff
@@ -28,16 +29,17 @@ pub enum ExecutionEvent {
 /// Management commands sent to the `execution` component.
 pub enum ExecutionManagementCommand {}
 
-#[derive(Debug, Clone)]
-struct SCELedgerEntry {
-    balance: Amount,
-    bytecode: Vec<u8>,
-    data: HHashMap<Hash, Vec<u8>>,
+/// execution request
+enum ExecutionRequest {
+    RunFinalStep(ExecutionStep),  // Runs a final step
```

Comment:
I think it would be better to have a `RunStep(ExecutionStep)`, where `ExecutionStep` is an enum differantiating between final and active.

Reply:
I wanted to make the concept of "execution step" finality-agnostic.

If you look at the run_final_step method, you'll see:

```rust
        self.reset_to_final();
        self.run_active_step(step.clone());
        self.run_final_step(step);
```